main.tf

```diff
@@ -1 +1,231 @@
-# main.tf placeholder
+#
+# Combine Policies
+#
+
+
+data "aws_iam_policy_document" "combinedPolicy" {
+
+  #
+  # Deny root account
+  #
+
+  statement {
+    sid       = "DenyRootAccount"
+    actions   = ["*"]
+    resources = ["*"]
+    effect    = local.deny_root_account_effect
+    condition {
+      test     = "StringLike"
+      variable = "aws:PrincipalArn"
+      values   = ["arn:aws:iam::*:root"]
+    }
+  }
+
+  #
+  # Deny leaving AWS Organizations
+  #
+
+  statement {
+    sid       = "DenyLeavingOrgs"
+    effect    = local.deny_leaving_orgs_effect
+    actions   = ["organizations:LeaveOrganization"]
+    resources = ["*"]
+  }
+
+  #
+  # Deny creating IAM users or access keys
+  #
+
+  statement {
+    sid    = "DenyCreatingIAMUsers"
+    effect = local.deny_creating_iam_users_effect
+    actions = [
+      "iam:CreateUser",
+      "iam:CreateAccessKey"
+    ]
+    resources = ["*"]
+  }
+
+
+
+  #
+  # Deny deleting KMS Keys
+  #
+
+  statement {
+    sid    = "DenyDeletingKMSKeys"
+    effect = local.deny_deleting_kms_keys_effect
+    actions = [
+      "kms:ScheduleKeyDeletion",
+      "kms:Delete*"
+    ]
+    resources = ["*"]
+  }
+
+  #
+  # Deny deleting Route53 Hosted Zones
+  #
+
+  statement {
+    sid    = "DenyDeletingRoute53Zones"
+    effect = local.deny_deleting_route53_zones_effect
+    actions = [
+      "route53:DeleteHostedZone"
+    ]
+    resources = ["*"]
+  }
+
+  #
+  # Deny all access
+  #
+
+  statement {
+    sid       = "DenyAllAccess"
+    effect    = local.deny_all_access_effect
+    actions   = ["*"]
+    resources = ["*"]
+  }
```

Comment:
We talked about this earlier -- this should be in a separate policy document, and then we want a switch to use either the one with all the other policies, or this one.